File: DesignPattern/DesignPattern/Decorator/BigTrouser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DesignPattern.Decorator
{
    class BigTrouser:Finery
    {
        public override void Show()
        {
            Console.WriteLine("跨裤！");
            base.Show();
        }
    }
}

File: DesignPattern/DesignPatternGo/decorators/bigtrouser.go
package decorators

type BigTrouser struct {
	finery Finery
	name   string
}

func (bt BigTrouser) Show() string {
	return bt.name + "、" + bt.finery.Show()
}

func NewBigTrouser(finery Finery) BigTrouser {
	return BigTrouser{
		name:   "大裤子",
		finery: finery,
	}
}
